web/controllers/users_controller.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Notox.UserController do
+  use Notox.Web, :controller
+  alias Notox.User
+
+  def index(conn, _params) do
+    users = Repo.all(User)
+    render conn, :index, users: users
+  end
+
+  def show(conn, %{"id" => id}) do
+    user = Repo.get(User, id)
+    render conn, :show, user: user
+  end
+
+  def new(conn, _params) do
+    changeset = User.changeset(%User{})
+    render conn, :new, changeset: changeset
+  end
+
+  def create(conn, %{"user" => user_params}) do
+    changeset = User.registration_changeset(%User{}, user_params)
+    case Repo.insert(changeset) do
+      {:ok, user} ->
+        conn
+        |> Notox.Auth.login(user)
+        |> put_flash(:info, "User created!")
+        |> redirect(to: user_path(conn, :index))
+      {:error, changeset} ->
+        render(conn, :new, changeset: changeset)
+    end
+  end
+
+  def edit(conn, %{"id" => id}) do
```

Comment:
Hmm, I probably wouldn't use `profile` for deleting the current user. I think of a profile as the data associated with a user. Things like name, bio, likes, avatar. It makes sense to edit and view a profile, but probably not to delete one.

As for a profile model, if you had a lot of data/logic around a profile it _might_ make sense to break that information away into it's own model, but I probably would start with _just_ a user model and allow it to grow naturally.

In the case of deleting the current user, I could see that going in an `account` controller, which would lead me to believe that both of these things should go in an account controller for now.

There isn't a hard and fast rule for any of this. It's just a communication/organization tool, and it can be hard to pick just the right thing. Sometimes it becomes more apparent as the app grows.